### PR TITLE
fix: Display current user's ENS in profile and share URL

### LINKED ISSUE #2503: ENS name is not being added to Profile when having ENS name set in settings
since the ENS name is not being set in Profile when it is set in settings, it causes a lot of issues when chatting / transacting with mobile, please assist with fixing the issue

**Steps:**

1. create account
2. buy ENS name in ENS names dapp 
3. set the ens name as primary
4. check the profile

As a result, ens name is not replacing 3-words-name even it is being set as primary name

`DBG 2021-06-03 12:55:01+03:00 Verifying mailserver connection state...   topics="mailserver model" tid=2931892 file=model.nim:186
qrc:/app/AppLayouts/Chat/ChatColumn/MessageComponents/UserImage.qml:9:5: Unable to assign [undefined] to QString
qrc:/app/AppLayouts/Chat/ChatColumn/MessageComponents/UsernameLabel.qml:43:9: Unable to assign [undefined] to QString
qrc:/app/AppLayouts/Chat/ChatColumn/MessageComponents/UsernameLabel.qml:42: ReferenceError: userName is not defined
qrc:/app/AppLayouts/Chat/ChatColumn/MessageComponents/ChatTime.qml:10: ReferenceError: isImage is not defined`

Screenshot 2021-06-03 at 12 17 28

Screenshot 2021-06-03 at 12 17 24

The idea is coming from mobile app: when having ENS name bought and set up in settings -> share profile URL looks like `https://join.status.im/u/tanyatest10.stateofus.eth`

Currently on desktop even having ENS name bought and set in profile, share your profile link looks as follows: `https://join.status.im/u/0x0461904235fb5762daf2970bd61b070784f3705fc51d5c96ea798d81a6552599c4f6c88ba9e83686080710b419dbdcca7a5c5ff9e8de7dd58f38a1c8be4c9a5a1d`

Would be great to implement the feature the way is is present on mobile, so once i buy ENS name and set some in Chat settings -> Primary Username -> Share Profile link will contain the selected ENS name. 

Clicking such link should behave the same as for the link with chat key -> sending to user profile.

@errorists @simonricoo

### PR DESCRIPTION
**Pull request by richard-ramos**
Fixes #2503

### CODE REVIEW

File: ui/app/AppLayouts/Chat/components/ProfilePopup.qml [hunk 1]
```diff
@@ -187,10 +188,36 @@ ModalPopup {
...
+                } else {
+                    if (isEnsVerified) {
+                        user = userName.startsWith("@") ? userName.substring(1) : userName
+                    }
+                }
+                if (user === ""){
+                    user = fromAuthor.substr(0, 4) + "..." + fromAuthor.substr(fromAuthor.length - 5)
+                }
+                return Constants.userLinkPrefix +  user;
+            }
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/ProfilePopup.qml:204]: The share link also works with the ENS name? TIL

**Review by jrainville — APPROVED** [review]

**Comment by iurimatias** [comment]
tagging with QA tag cc: @anastasiyaig

**Comment by anastasiyaig** [comment]
@richard-ramos

- [ ] clicking such a link https://join.status.im/u/anastasija.stateofus.eth in chat hangs the app (it becomes irresponsive), also i do see an error shown in console:

`ERR 2021-07-26 15:17:53+03:00 rpc response error                         topics="rpc" tid=11056733 file=core.nim:25 methodName=wakuext_markAllRead payload="[\"anastasija.stateofus.eth\"]" result="{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"code\":-32000,\"message\":\"chat not found\"}}"`

- [ ] when setting a ens name in profile, the design misses `Hey` word:

Screenshot 2021-07-26 at 15 27 49

- [ ] link preview has misleading wording

Screenshot 2021-07-26 at 15 32 25

- [ ] missing information when clicking the ens name

Screenshot 2021-07-26 at 15 42 23


can we fix something in this PR?

**Comment by richard-ramos** [comment]
I can fix all except:
> missing information when clicking the ens name

which should be done in a separate PR (that should fix https://github.com/status-im/status-desktop/issues/782 as well)

**Comment by anastasiyaig** [comment]
@richard-ramos thank you for fixing all this, i will log the leftover as a separate issue

**Pull request merged**